### ISSUE TITLE
E342: Out of memory! when closing a popup window with the mouse

### ISSUE THREAD

**Issue by bfrg** [vim/vim#4859]
**Describe the bug**
Since `:source` doesn't accept a range, I started using the following command to execute a selection:
```vim
command! -buffer -range Source silent <line1>,<line2>yank z | let @z = substitute(@z, '\n\s*\\', '', 'g') | @z
```
When a popup window was created through executing the content of a register, and the window is closed with the mouse, Vim throws an `E342: Out of memory!` error. 
The error doesn't occur when a mouse selection was made inside the popup window before it is closed.

**To Reproduce**
1. Run `vim --clean popup.vim -c "source popup.vim"`
```vim
" popup.vim
command! -buffer -range Source silent <line1>,<line2>yank z | let @z = substitute(@z, '\n\s*\\', '', 'g') | @z
call popup_create(['one', 'two', 'three four'], #{
    \ border: [1,1,1,1],
    \ close: 'button'
    \ })
```
2. Close the popup window with the mouse by clicking on `X`. No errors will appear because the script was sourced from the command line.
3. Run `:3,6Source`, and then close the popup window with the mouse.
```
E342: Out of memory!  (allocating 18446744073709547626 bytes)
```
4. Redo step 3, but before closing the popup window, first make a mouse selection inside the popup window. Then click on `X`. No error message will appear now.

**Screenshots**
There also seems to be a drawing issue in Gvim after the popup window was closed. This does not occur in terminal Vim:

![screenshot-2019-08-24_014133](https://user-images.githubusercontent.com/6266600/63629282-9222ab00-c611-11e9-91ce-da07ab6eb1ee.png)

**Environment**
- Vim version `8.1.1912`
- OS: Gentoo Linux
- Terminal: xfce4-terminal

**Comment by brammool** [comment]
> **Describe the bug**
> Since `:source` doesn't accept a range, I started using the following command to execute a selection:
> ```vim
> command! -buffer -range Source silent <line1>,<line2>yank z | let @z = substitute(@z, '\n\s*\\', '', 'g') | @z
> ```
> When a popup window was created through executing the content of a register, and the window is closed with the mouse, Vim throws an `E342: Out of memory!` error. 
> The error doesn't occur when a mouse selection was made inside the popup window before it is closed.
> 
> **To Reproduce**
> 1. Run `vim --clean popup.vim -c "source popup.vim"`
> ```vim
> " popup.vim
> command! -buffer -range Source silent <line1>,<line2>yank z | let @z = substitute(@z, '\n\s*\\', '', 'g') | @z
> call popup_create(['one', 'two', 'three four'], #{
>     \ border: [1,1,1,1],
>     \ close: 'button'
>     \ })
> ```
> 2. Close the popup window with the mouse by clicking on `X`. No errors will appear because the script was sourced from the command line.
> 3. Run `:3,6Source`, and then close the popup window with the mouse.
> ```
> E342: Out of memory!  (allocating 18446744073709547626 bytes)
> ```
> 4. Redo step 3, but before closing the popup window, first make a mouse selection inside the popup window. Then click on `X`. No error message will appear now.
> 
> **Screenshots**
> There also seems to be a drawing issue in Gvim after the popup window was closed. This does not occur in terminal Vim:
> 
> ![screenshot-2019-08-24_014133](https://user-images.githubusercontent.com/6266600/63629282-9222ab00-c611-11e9-91ce-da07ab6eb1ee.png)

Thanks for making a reproducible example.  I had seen the problem
before, but haven't managed to trigger it.  Now I can fix it.

-- 
hundred-and-one symptoms of being an internet addict:
104. When people ask about the Presidential Election you ask "Which country?"

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by bfrg** [comment]
Thank you. I think this was fixed in commit 741ea170 and can be closed. I can't reproduce it anymore.
